mvcc-rs/src/database/mod.rs

```diff
@@ -784,18 +797,18 @@ pub(crate) fn is_write_write_conflict(
     }
 }
 
-pub(crate) fn is_version_visible(
+pub(crate) fn is_version_visible<T>(
     txs: &SkipMap<TxID, RwLock<Transaction>>,
     tx: &Transaction,
-    rv: &RowVersion,
+    rv: &RowVersion<T>,
 ) -> bool {
     is_begin_visible(txs, tx, rv) && is_end_visible(txs, tx, rv)
 }
 
-fn is_begin_visible(
+fn is_begin_visible<T>(
```

Comment:
In a follow-up commit, we should consider detemplatizing these functions. They don't really depend on `T`, so perhaps we can simply add a `VersionTimestamps` struct that has `{ begin: TxTimestampOrID, end: Option<TxTimestampOrID>}` fields, with `RowVersion` implementing `Defer<VersionTimestamps>`. Otherwise the compiled code will get bloated if we instantiate lots of different `T`s. That's a future consideration though, it's fine as is for now